Naff

LordOfPolls opened the pull request:
## What type of pull request is this?

<!-- Check whichever applies to your PR -->
- [ ] Non-breaking code change
- [x] Breaking code change
- [ ] Documentation change/addition
- [ ] Tests change

## Description
<!-- Clearly and concisely describe what this PR is for, and why you feel it should be merged. -->
I can not exagerate how breaking this is. Your code will not work at all from this code base. 

This renames everything from snake, or snake like naming to `Naff`, as well as improving general performance of the libary. While `dis-snek` pends its final release, this will branch will act as a staging branch for the library

## Changes
<!-- - A bullet pointed list outlining the changes you have made -->
- Renames `Snake` client -> `Client` 
- Renames `Scale` to `Cog`
- Renames `models/snek` -> `models/naff`
- Renames `SnakeException` -> `NaffException`
- Renames `SnakeBotUser` -> `NaffUSer`
- Renames `grow_scale` -> `mount_cog`
- Renames `shed_scale` -> `drop_cog`
- Renames `get_scales` -> `Client.get_cogs`
- Renames `get_scale` -> `Client.get_cogs`
- Renames `Snake.sync_scales` -> `Client.sync_cogs`
- Renames `Snake.scales` -> `Clientcogs`
- Renames `Command.scale` -> `Command.cog`
- Renames `Scale.*scale*` -> `Cog.*cog*`
- Renames `raw_socket_event` -> `raw_gateway_event`
Probably a load more renames
- Add automated testing bot 


## Checklist

<!-- These are actions you **must** have taken, if you haven't, your PR will be rejected -->
- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

LordOfPolls committed:
refactor: Rename root folder

LordOfPolls committed:
refactor: rename Snake -> client

LordOfPolls committed:
refactor: rename SnakeBotUser -> NaffUser

LordOfPolls committed:
refactor: rename task name for dispatcher

LordOfPolls committed:
refactor: Rename SnakeException -> NaffException

LordOfPolls committed:
docs: Remove client snake comments

LordOfPolls committed:
refactor: rename models/snek -> models/naff

LordOfPolls committed:
refactor: Rename Scale -> Cog

LordOfPolls committed:
refactor: Remove snek/snake references

LordOfPolls committed:
tests: refactor tests

LordOfPolls committed:
build: rename library

LordOfPolls committed:
docs: update docs for new structure

LordOfPolls committed:
Merge remote-tracking branch 'origin/dev' into naff

zevaryx reviewed (approved):
Okay, @LordOfPolls , look. It's 228 changed files and 2000 changes. I don't have the time to manually review this. However, I am running my main bot on this branch as of an hour ago and everything is working as intended, so I tested it and verified that it's working.

LordOfPolls committed:
build: update build information

LordOfPolls committed:
perf: Add Brotli speedup extra dep

LordOfPolls committed:
perf: use tuples for context menu cache lookup

LordOfPolls committed:
perf: import methods directly in player

LordOfPolls committed:
perf: use list comprehension for thread parsing

LordOfPolls committed:
perf: Use tuple instead of list

LordOfPolls committed:
perf: Convert for loops in client into comprehensions where applicable

LordOfPolls committed:
perf: Convert const list into tuple

Consts are... constant, use a tuple

LordOfPolls committed:
perf: another comprehension conversion

LordOfPolls committed:
fix💥: Rename `raw_socket_event` -> `raw_gateway_event`

LordOfPolls committed:
refactor: Fix erroneous dis-naff mistake

LordOfPolls committed:
tests: Add automated tests bot (#468)

* tests: Merge snek-tests into lib pytests

* test: install voice deps to env

* tests: skip voice on non-voice extra

* fix: Wait for member update event

* tests: Use pyproj for coverage

* tests: Remove now redundant pragmas

* tests: prevent multiple test-bots running concurrently

LordOfPolls committed:
Merge remote-tracking branch 'origin/dev' into naff

LordOfPolls committed:
tests: update imports to naff

LordOfPolls committed:
fix: Fix weird caching behaviour of `get_init_keys`

LordOfPolls committed:
Merge branch 'dev' into naff

LordOfPolls committed:
Merge branch 'dev' into naff

i0bs committed:
feat💥: convert `cover_image` in `Application` to `Asset`. (#475)

The `cover_image` is currently handled as a string of its image hash. Since this attribute in the Discord Developers documentation shares the [same CDN endpoint as the `icon` attribute,](https://i.imgur.com/G0fXSfq.png) I've made a quick edit to its typing and altered the `_process_dict` classmethod to reflect it.

i0bs committed:
feat: expose `Application` attrs for App Discovery (#476)

* feat: expose `Application` attrs for App Discovery

This simply exposes the `tags`, `install_params` and `custom_install_url` which were recently amended to the `Application` object when application discovery was announced. These also may have just been simply overlooked

* feat: add `install_params` attr

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: LordOfPolls <ddavidallen13@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

LordOfPolls committed:
feat💥: Support auto-mod (#477)

* feat: Add auto-moderation audit log types

* refactor💥: Rename message type `auto_mod` to match api

the api refers to it as AUTO_MODERATION_ACTION  not the placeholder of `AUTO_MOD`

LordOfPolls committed:
Merge branch 'dev' into naff

i0bs committed:
docs: correct grammar in `ThreadableMixin`. (#481)

This correct a typo and grammatical error in the docstring for `ThreadableMixin` and argument signature.

i0bs committed:
docs: create admonition warnings for components. (#482)

This simply adds warning admonitions for the `BaseComponent` and `InteractiveComponent` classes to provide a more appealing UI for making it clear they *shouldn't* be ever instantiated.

Kigstn reviewed (approved):
Have been running this without problems

LordOfPolls committed:
Merge branch 'dev' into naff
